warp/aggregator/aggregation_job_test.go

```diff
@@ -353,3 +354,56 @@ func TestAggregateThresholdSignaturesParentCtxCancels(t *testing.T) {
 		expectedErr: avalancheWarp.ErrInsufficientWeight,
 	})
 }
+
+func TestAggregateThresholdSignaturesParentCtxCancelsAfterDelay(t *testing.T) {
+	// Get one signature exactly and then parent context cancels
```

Comment:
could we build this behavior deterministically instead of relying on the closeness of 1 second to 1001 ms (and also without slowing down the test for 1s)

Reply:
I can reduce the test length, but not sure how else I can deterministically cancel the parent context in the middle of the test without using timers. I don't exactly know when the test grabs a signature (unless we added channels to report that)

Reply:
Using a timer to block fetching signatures until it fires is the closest thing I have to approximating the function time.